workers/unity/Packages/com.improbable.gdk.transformsynchronization/TransformUtils.cs

```diff
@@ -11,51 +11,146 @@ public static bool HasChanged(Coordinates a, Coordinates b)
         }
 
         // Checking for no change, so exact equality is fine
-        public static bool HasChanged(Location a, Location b)
+        public static bool HasChanged(CompressedQuaternion a, CompressedQuaternion b)
         {
-            return a.X != b.X || a.Y != b.Y || a.Z != b.Z;
+            return a.Data != b.Data;
         }
 
         // Checking for no change, so exact equality is fine
-        public static bool HasChanged(Quaternion a, Quaternion b)
+        public static bool HasChanged(FixedPointVector3 a, FixedPointVector3 b)
+        {
+            return a.X != b.X || a.Y != b.Y || a.Z != b.Z;
+        }
+
+        // Decompress from uint32 to Unity Quaternion
+        public static UnityEngine.Quaternion ToUnityQuaternion(this CompressedQuaternion quaternion)
+        {
+            var compressedValue = quaternion.Data;
+            var q = new float[4];
+
+            const uint mask = (1u << 9) - 1u;
```

Comment:
Why 9

Reply:
If you want to represent 0 exactly generally you can map from the quaternion float range to the fixed int range -1. You don't need to lose an entire bit.

Reply:
Actually now that I think about it that's the only way of doing it. 2^n is always even :P

Reply:
atm the identity quaternion (0,0,0,1) gets packed to [3 0 0 0] (index 3, value 0, value 0, value 0)

wouldn't it be easier to make 0 the identity by switching the order in compression to wxyz, so the packed structure would become [0 0 0 0]?

Reply:
I don't understand. If you want 0 to be the identity then special case it.